### PR TITLE
DataLogger: replace `DataLogger` with `Experimental.DataStreamer2`

### PR DESCRIPTION
**Pull request by drolevar**
This should be enough to replace DataLogger with DataStreamer2.

The existing use cases should not be broken.

I've decided to put the new functionality into DataLogger and not DataStreamer because DataStreamer2 is actually based on DataLogger and the API is compatible.

### CODE REVIEW

**Comment by drolevar** [comment]
@swharden Any comments on this PR?

**Comment by swharden** [comment]
Hi @drolevar, this looks great! 

I was out of town for about a week then got behind on some other stuff so I've been away from ScottPlot for a few weeks and just now catching up 😅 

> The existing use cases should not be broken

This certainly makes things easier. Thanks for implementing this in a non-breaking way! I'm excited to see this will be in the next release 👍

File: src/ScottPlot5/ScottPlot5/Plottables/DataLogger.cs [hunk 1]
```diff
@@ -1,141 +1,239 @@
...
-        Data.Add(x, y);
+        CoordinateRange rangeX = Data.GetRangeX();
+        CoordinateRange rangeY = Data.GetRangeY(rangeX);
+
+        return Rotated
+            ? new AxisLimits(rangeY, rangeX)
+            : new AxisLimits(rangeX, rangeY);
     }
 
-    public void Add(double[] xs, double[] ys)
```
Comment (jpgarza93) [inline src/ScottPlot5/ScottPlot5/Plottables/DataLogger.cs:62]: Why was this removed? This was supper useful when loading data from a csv file
